diff --git a/sycl/plugins/level_zero/ze_api_generator.py b/sycl/plugins/level_zero/ze_api_generator.py
index ca70341c93a4..b49060beb7bf 100644
--- a/sycl/plugins/level_zero/ze_api_generator.py
+++ b/sycl/plugins/level_zero/ze_api_generator.py
@@ -35,6 +35,6 @@ def extract_ze_apis(header):
     api.close()
 
 if __name__ == "__main__":
-    with open(sys.argv[1], 'r') as f:
+    with open(sys.argv[1], 'r', encoding='utf-8') as f:
         header = f.read()
         extract_ze_apis(header)
diff --git a/sycl/tools/sycl-trace/generate_pi_pretty_printers.py b/sycl/tools/sycl-trace/generate_pi_pretty_printers.py
index 07b0c3a41c94..f290c73414c9 100644
--- a/sycl/tools/sycl-trace/generate_pi_pretty_printers.py
+++ b/sycl/tools/sycl-trace/generate_pi_pretty_printers.py
@@ -49,6 +49,6 @@ if __name__ == "__main__":
     """
     Usage: python generate_pi_pretty_printers.py path/to/pi.h
     """
-    with open(sys.argv[1], 'r') as f:
+    with open(sys.argv[1], 'r', encoding='utf-8') as f:
         header = f.read()
         generate_pi_pretty_printers(header)
diff --git a/sycl/tools/sycl-trace/generate_ze_pretty_printers.py b/sycl/tools/sycl-trace/generate_ze_pretty_printers.py
index 40d1b600b9c5..f4df492c6ea5 100644
--- a/sycl/tools/sycl-trace/generate_ze_pretty_printers.py
+++ b/sycl/tools/sycl-trace/generate_ze_pretty_printers.py
@@ -53,7 +53,7 @@ if __name__ == "__main__":
     """
     Usage: python generate_pi_pretty_printers.py path/to/ze_api.h
     """
-    with open(sys.argv[1], 'r') as f:
+    with open(sys.argv[1], 'r', encoding='utf-8') as f:
         header = f.read()
         generate_ze_pretty_printers(header)
 
